src/renderer/services/wallet/util.ts

```diff
@@ -17,3 +22,40 @@ export const hasKeystoreContent = (state: KeystoreState): boolean => isSome(getK
 export const hasImportedKeystore = (state: KeystoreState): boolean => isSome(state)
 
 export const isLocked = (state: KeystoreState): boolean => !hasImportedKeystore(state) || !hasKeystoreContent(state)
+
+export const assetWithBalanceMonoid = getMonoid<AssetWithBalance>()
+
+export const filterNullableBalances = (balances: AssetsWithBalance) => {
+  return FP.pipe(
+    balances,
+    A.filter((balance) => balance.amount.amount().isPositive())
+  )
+}
+
+const TICKERS_ORDER = ['BTC', 'RUNE', 'BNB']
+
+const getBalanceIndex = (balance: AssetWithBalance) =>
+  TICKERS_ORDER.findIndex((ticker) => ticker === balance.asset.ticker)
+
+const byTickersOrder = Ord.ord.contramap(Ord.ordNumber, getBalanceIndex)
+
+// We will compare asset strings and they automatically
+// be grouped by their chains in alphabetic order
+const byAsset = Ord.ord.contramap(Ord.ordString, (balance: AssetWithBalance) => assetToString(balance.asset))
+
+export const sortBalances = (balances: AssetsWithBalance) => {
+  return FP.pipe(
+    balances,
+    // split array for 2 parts: sortable assets and the rest
+    A.reduce([[], []] as [AssetsWithBalance, AssetsWithBalance], (acc, cur) => {
```

Comment:
[suggestion] How about to reduce it into `Either<AssetsWithBalance, AssetsWithBalance>` instead into an array with two elements `[[], []]`?

Reply:
yeah. i thought about it too. but actually it does not make sense in this case. anyway we need to get an array as a result for this function. in case of `Either` we would need to store `Either[]` 'cause Either can store only a single value (left OR right) and after that we would need to split that array for 2 parts again and map each part as `Either` values.
in this case here is no problems with using such array. such type definition allows to define ALL array element and attempt to read outer index will be restricted by compiler
![image](https://user-images.githubusercontent.com/67793194/92625876-dee72b80-f2d1-11ea-9b46-99ecb2a5e2e2.png)
all nested arrays are processed in a 'safe mode'

Reply:
You're right, we still need two arrays ...